### ISSUE TITLE
Direct link to Q&A

### ISSUE THREAD

**Issue by nimdanor** [PremierLangage/premierlangage#246]
We need to create the ability to create an LTI link directly to Q & A.
And a link to a course that will limit questions and answers about the course using a key word equal to the course name.

But if there is the question is general, we should find it anyway.

This must be discussed in the monday breefing.

**Comment by nimdanor** [comment]
Are there more interfaces in the Q&A,
is there a moderator view ?

**Comment by qcoumes** [comment]
Why do you mean by interfaces ?

Admin have the same view, but should be able to edit/delete any question/answer/comment without the need for reputation.

**Comment by nimdanor** [comment]
I need a dsicription of the possibilities of the installed software. I don't know the possibilites.

i would like to give administrtive privilege to people on a restricted part of the Q&A,
How is tis possible with the software currently installed ?

**Comment by qcoumes** [comment]
It's simple Q&A where right are tied to a reputation system. Reputation gain for actions and reputation needed to do something are set in `config.py`, or if absent, use these fault : 

```python
# Ask's settings
# Reputation contains the points awarded
# Right contains the points needed for doing action, -1 to disable (do not apply for the owner)
QA_SETTINGS = {
    'qa_messages'            : True,
    'qa_description_optional': False,
    'reputation'             : {
        'CREATE_QUESTION'         : 10,
        'CREATE_ANSWER'           : 20,
        'CREATE_ANSWER_COMMENT'   : 2,
        'CREATE_QUESTION_COMMENT' : 2,
        'RECEIVE_QUESTION_COMMENT': 1,
        'RECEIVE_ANSWER_COMMENT'  : 1,
        'ANSWER_ACCEPTED'         : 20,  # Half for the acceptor
        'UPVOTE_QUESTION'         : 3,
        'UPVOTE_ANSWER'           : 3,
        'DOWNVOTE_QUESTION'       : -3,
        'DOWNVOTE_ANSWER'         : -3,
    },
    'right'                  : {
        'POST_QUESTION'  : 0,
        'POST_ANSWER'    : 0,
        'POST_COMMENT'   : 0,
        'EDIT_QUESTION'  : 500,
        'EDIT_ANSWER'    : 500,
        'EDIT_COMMENT'   : -1,
        'DELETE_QUESTION': 2000,
        'DELETE_ANSWER'  : 2000,
        'DELETE_COMMENT' : 2000,
    },
}
```
Meaning that, by default, anyone can ask a question, answer and post a comment.

Every questions, answers and comments are visible by everyone.

___

So it is not possible to have a restricted part on the current Q&A.

**Comment by nimdanor** [comment]
Are  there a systeme of tags ?
if there is we could allways associate a tag to question asked in a given course?

**Comment by nimdanor** [comment]
how can i be an q&a admin on the prod ?

**Comment by qcoumes** [comment]
Yes, there's a system of tag, [here for instance all question related to recusivity on pl-preprod](https://pl-preprod.u-pem.fr/ask/?tag=recursivite). Currently, every question should have at least one tag.

About the roles on Q&A, they're currently tied to the role of PL. So you need to be admin on PL to be admin on Q&A.

We lack system of group and permissions to allow differents roles on different parts of the website.

**Comment by nimdanor** [comment]
Let's have groups ;)

**Comment by nimdanor** [comment]
Depends pon 247

**Comment by nimdanor** [comment]
@nimdanor couxcou
